atom/renderer/atom_sandboxed_renderer_client.cc

```diff
@@ -0,0 +1,201 @@
+                    const base::string16& channel,
+                    const base::ListValue& args) override {
+    if (!frame || frame->isWebRemoteFrame())
+      return;
+
+    auto isolate = blink::mainThreadIsolate();
+    v8::HandleScope handle_scope(isolate);
+    auto context = frame->mainWorldScriptContext();
+    v8::Context::Scope context_scope(context);
+    v8::Local<v8::Value> argv[] = {
+      mate::ConvertToV8(isolate, channel),
+      mate::ConvertToV8(isolate, args)
+    };
+    renderer_client_->InvokeBindingCallback(
+        context,
+        "onMessage",
+        std::vector<v8::Local<v8::Value>>(argv, argv + 2));
+  }
+
+ private:
+  AtomSandboxedRendererClient* renderer_client_;
+  DISALLOW_COPY_AND_ASSIGN(AtomSandboxedRenderViewObserver);
+};
+
+}  // namespace
+
+
+AtomSandboxedRendererClient::AtomSandboxedRendererClient() {
+}
+
+AtomSandboxedRendererClient::~AtomSandboxedRendererClient() {
+}
+
+void AtomSandboxedRendererClient::RenderFrameCreated(
+    content::RenderFrame* render_frame) {
+  new AtomSandboxedRenderFrameObserver(render_frame, this);
+}
+
+void AtomSandboxedRendererClient::RenderViewCreated(
+    content::RenderView* render_view) {
+  new AtomSandboxedRenderViewObserver(render_view, this);
+}
+
+void AtomSandboxedRendererClient::DidCreateScriptContext(
+    v8::Handle<v8::Context> context, content::RenderFrame* render_frame) {
+  base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
+  std::string preload_script = command_line->GetSwitchValueASCII(
+      switches::kPreloadScript);
+  if (preload_script.empty())
+    return;
+
+  auto isolate = context->GetIsolate();
+  v8::HandleScope handle_scope(isolate);
+  v8::Context::Scope context_scope(context);
+  // Wrap the bundle into a function that receives the binding object and the
+  // preload script path as arguments.
+  std::string preload_bundle_native(node::preload_bundle_native,
+      node::preload_bundle_native + sizeof(node::preload_bundle_native));
+  std::stringstream ss;
+  ss << "(function(binding, preloadPath) {\n";
+  ss << preload_bundle_native << "\n";
+  ss << "})";
+  std::string preload_wrapper = ss.str();
+  // Compile the wrapper and run it to get the function object
+  auto script = v8::Script::Compile(v8::String::NewFromUtf8(
+        isolate,
+        preload_wrapper.c_str(),
+        v8::String::kNormalString,
+        preload_wrapper.length()));
+  auto func = v8::Handle<v8::Function>::Cast(
+      script->Run(context).ToLocalChecked());
+  // Create and initialize the binding object
+  auto binding = v8::Object::New(isolate);
+  api::Initialize(binding, v8::Null(isolate), context, nullptr);
+  v8::Local<v8::Value> args[] = {
+    binding,
+    v8::String::NewFromUtf8(isolate, preload_script.c_str(),
+        v8::NewStringType::kNormal).ToLocalChecked()
+  };
+  // Execute the function with proper arguments
+  (void)func->Call(context, v8::Null(isolate), 2, args);
```

Comment:
There is a `ignore_result` helper to ignore results, the `(void)` hack should be avoided.